pkg/ui/workspaces/cluster-ui/src/api/databases/tableMetaUpdateJobApi.ts

```diff
@@ -0,0 +1,141 @@
+// Copyright 2024 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+import moment from "moment-timezone";
+import { useEffect, useMemo } from "react";
+import useSWR from "swr";
+
+import { fetchDataJSON } from "../fetchData";
+
+const TABLE_META_JOB_API = "api/v2/table_metadata/updatejob/";
+
+export enum TableMetadataJobStatus {
+  NOT_RUNNING = "NOT_RUNNING",
+  RUNNING = "RUNNING",
+}
+
+type TableMetaUpdateJobResponse = {
+  current_status: TableMetadataJobStatus;
+  progress: number;
+  last_start_time?: string;
+  last_completed_time: string;
+  last_updated_time: string;
+  data_valid_duration: number;
+  automatic_updates_enabled: boolean;
+};
+
+type TableMetaUpdateJobInfo = {
+  currentStatus: TableMetaUpdateJobResponse["current_status"];
+  progress: number;
+  lastStartTime: moment.Moment;
+  lastCompletedTime: moment.Moment;
+  lastUpdatedTime: moment.Moment;
+  dataValidDuration: moment.Duration;
+  automaticUpdatesEnabled: boolean;
+};
+
+const getTableMetaUpdateJobInfo = async () => {
+  return fetchDataJSON<TableMetaUpdateJobResponse, null>(TABLE_META_JOB_API);
+};
+
+// useTableMetaUpdateJob is a hook that fetches the current status of the table
+// metadata update job and returns the status and other relevant information.
+// The hook polls the API every 3 seconds if the job status is parsed as running.
+export const useTableMetaUpdateJob = () => {
+  const { data, isLoading, mutate } = useSWR(
+    "tableMetaUpdateJob",
+    getTableMetaUpdateJobInfo,
+    {
+      focusThrottleInterval: 10000,
+      refreshInterval: (latest: TableMetaUpdateJobResponse) => {
+        return latest?.current_status === TableMetadataJobStatus.RUNNING
+          ? 3000
+          : 0;
+      },
+      dedupingInterval: 3000,
+    },
+  );
+
+  const formattedResp: TableMetaUpdateJobInfo = useMemo(() => {
+    if (!data) {
+      return null;
+    }
+    return {
+      currentStatus: data.current_status,
+      progress: data.progress,
+      lastStartTime: data.last_start_time
+        ? moment.utc(data.last_start_time)
+        : null,
```

Comment:
what's going on here? why is it only sometimes wrapped in `moment`?

Reply:
It's only converted to moment when the `lastStartTime` is is not null or empty.